node/src/test_utils/recorder_stop_conditions.rs

```diff
@@ -124,6 +124,13 @@ macro_rules! match_every_type_id{
     }
 }
 
+#[macro_export]
+macro_rules! match_every_type_id_partial_eq_less {
+    ($($single_message: ident),+) => {
```

Comment:
This makes me wonder if you can attempt to downcast `$single_message` into `PartialEq`, and wrap it in `PretendMatchable` if you fail. Then you might not need this macro.